addons/point_of_sale/static/tests/unit/related_models_test.js

```diff
@@ -0,0 +1,948 @@
+        });
+    });
+
+    QUnit.module("many2many field relations to other models", (hooks) => {
+        let models;
+        hooks.beforeEach(() => {
+            [models] = createRelatedModels({
+                "product.product": {
+                    name: { type: "char" },
+                    tag_ids: {
+                        type: "many2many",
+                        relation: "product.tag",
+                        relation_table: "product_tag_product_product_rel",
+                    },
+                },
+                "product.tag": {
+                    name: { type: "char" },
+                    product_ids: {
+                        type: "many2many",
+                        relation: "product.product",
+                        relation_table: "product_tag_product_product_rel",
+                    },
+                },
+            });
+        });
+
+        QUnit.test("create operation, create", (assert) => {
+            const tag1 = { name: "Electronics" };
+            const tag2 = { name: "New" };
+            const product = models["product.product"].create({
+                name: "Smartphone",
+                tag_ids: [["create", tag1, tag2]],
+            });
+            assert.ok(product.tag_ids[0].name, tag1.name, "tag1 should be linked with product");
+        });
+
+        QUnit.test("create operation, link", (assert) => {
+            const tag1 = models["product.tag"].create({ name: "Electronics" });
+            const tag2 = models["product.tag"].create({ name: "New" });
+            const product = models["product.product"].create({
+                name: "Smartphone",
+                tag_ids: [["link", tag1, tag2]],
+            });
+            assert.ok(product.tag_ids.includes(tag1), "tag1 should be linked with product");
+            assert.ok(
+                tag1.product_ids.includes(product),
+                "Product should be linked with both tags"
+            );
+        });
+
+        QUnit.test("read operation", (assert) => {
+            const p1 = models["product.product"].create({});
+            const p2 = models["product.product"].create({});
+            const p3 = models["product.product"].create({});
+            const t1 = models["product.tag"].create({ product_ids: [["link", p1, p2, p3]] });
+            const t2 = models["product.tag"].create({ product_ids: [["link", p1, p2]] });
+            // Test reading back the tags directly
+            const readT1 = models["product.tag"].read(t1.id);
+            assert.deepEqual(readT1, t1, "Tag 1 should be found and match the created tag");
+            const readP1 = models["product.product"].read(p1.id);
+            assert.deepEqual(readP1, p1, "Product should be found");
+            // Test the many2many relationship
+            assert.ok(
+                readT1.product_ids.includes(p1, p2, p3),
+                "Tag 1 should include Product 1, 2 and 3"
+            );
```

Comment:
```suggestion
            assert.ok(readT1.product_ids.includes(p1), "Product 1 should belong to Tag 1");
            assert.ok(readT1.product_ids.includes(p2), "Product 2 should belong to Tag 1");
            assert.ok(readT1.product_ids.includes(p3), "Product 3 should belong to Tag 1");
```

Reply:
Do this in other places where you provided multiple items when calling `includes`. That method only works for one item.